lib/srv/db/watcher.go

```diff
@@ -171,8 +171,19 @@ func (s *Server) matcher(resource types.ResourceWithLabels) bool {
 	if !ok {
 		return false
 	}
-	if database.IsRDS() || database.IsRedshift() {
+
+	// In the case of CloudOrigin RDS or Redshift DB resources the matchers should be skipped.
+	if hasOriginCloudLabel(resource) && (database.IsRDS() || database.IsRedshift()) {
 		return true // Cloud fetchers return only matching databases.
 	}
 	return services.MatchResourceLabels(s.cfg.ResourceMatchers, database)
 }
+
+func hasOriginCloudLabel(r types.ResourceWithLabels) bool {
+	for k, v := range r.GetAllLabels() {
+		if k == types.OriginLabel && v == types.OriginCloud {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
nit: `r.Origin() == types.OriginCloud` should also work?

Reply:
Yes, it works. Thanks for the suggestion.